spec/biker_spec.rb

```diff
@@ -0,0 +1,127 @@
+require "./lib/ride.rb"
+require "./lib/biker.rb"
+
+RSpec.describe Biker do
+    before(:all) do 
+        @biker1 = Biker.new("Kenny", 30)
+        @biker2 = Biker.new("Athena", 15)
+        @ride1 = Ride.new({name: "Walnut Creek Trail", distance: 10.7, loop: false, terrain: :hills})
+        @ride2 = Ride.new({name: "Town Lake", distance: 14.9, loop: true, terrain: :gravel})
+    end
+
+    describe "#initialize" do
+        it "can initialize" do
+            expect(@biker1).to be_an_instance_of (Biker)
+        end
+
+        it "can initialize wtih a name" do
+            expect(@biker1.name).to eq('Kenny')
+        end
+
+        it "can initialize with a max distance" do
+            expect(@biker1.max_distance).to eq(30)
+        end
+
+        it "can initialize with a rides" do
+            expect(@biker1.rides).to eq({})
+        end
+
+        it "can initialize with a rides" do
+            expect(@biker1.acceptable_terrain).to eq([])
+        end
+    end
+
+    describe "#learn_terrain!" do
+        it "can learn new terrain types" do
+            expect(@biker1.acceptable_terrain).to eq([])
+
+            @biker1.learn_terrain!(:gravel)
+
+            expect(@biker1.acceptable_terrain).to eq([:gravel])
+
+            @biker1.learn_terrain!(:hills)
+
+            expect(@biker1.acceptable_terrain).to eq([:gravel, :hills])
+
+            @biker1.learn_terrain!(:gravel)
+
+            expect(@biker1.acceptable_terrain).to eq([:gravel, :hills])
```

Comment:
Great test! I like that you tested that it could add a second item, but then it could not add the dup